### PR TITLE
Platforms to toolchains, CrossRules are no more

### PR DESCRIPTION
**Pull request by ptia**
Based on https://github.com/daedaleanai/dbt-rules/pull/18. Quite a big simplification:
* No more custom libsupcxx/RULES, the default rules are enough!
* The targets too are much simpler to write and read.

### CODE REVIEW

**Review by jfrohnhofen — APPROVED** [review]

**Review by ljanyst — APPROVED** [review]
I like it! Good job!

File: boot/linux-x86_64/BUILD.go [hunk 1]
```diff
@@ -5,12 +5,19 @@ import (
...
 	"dbt-rules/RULES/util"
 
-	"libsupcxx/RULES/platform"
+	"libsupcxx"
 	"libsupcxx/libsupcxx/include"
 
 	gcc "x86_64-elf-gcc"
 )
 
+var bootToolchain = gcc.Toolchain.NewWithStdLib(
```
Comment (ptia) [inline boot/linux-x86_64/BUILD.go:14]: This probably needs an explanation. I want the boot libs to be compiled with the same toolchains as the other libs (before they used the one from the gcc package, which is pretty weird). The problem is that if I try to say `Toolchain: Toolchain` to make sure that only the right toolchain is used, go complains that the `Toolchan` struct and the `boot*` libs are circularly dependent. This bootToolchain is used to fix that, breaking the circular dependency.

Comment (ptia) [reply boot/linux-x86_64/BUILD.go:14]: > The whole point is that they should not use standard headers!

I meant the std headers that come from libsupcxx, i.e. the same ones as the other libs.

Comment (ptia) [reply boot/linux-x86_64/BUILD.go:14]: (I can see where the confusion came from, I will change the commit comment for future reference before merging)

**Comment by ptia** [comment]
TODO: update dbt-rules version

**Pull request merged**